lib/payment.js

```diff
@@ -57,25 +58,36 @@ class Payment {
      * @param {Address} sender - Senders address
      * @param {Address} recipient - Recipient address
      * @param {Wallet|NahmiiProvider} [walletOrProvider] - An optional Wallet or NahmiiProvider instance
+     * @param {String} [senderPayload] - Optional stringified payment sender data payload
      * @param {String} [senderRef] - Optional uuid identifying the payment. Must be unique per sender wallet. Random if undefined.
      *
      * @example
-     * // Normal, random sender reference
+     * // Normal, no sender payload and random sender reference
      * const senderWallet = new nahmii.Wallet(...);
      * const recipientWallet = new nahmii.Wallet(...);
      * const paymentAmount = nahmii.MonetaryAmount.from(...);
      * const payment = new nahmii.Payment(paymentAmount, senderWallet.address, recipientWallet.address, senderWallet);
      *
-     * // Advanced, semantic sender reference. See: https://github.com/uuidjs/uuid
+     * // Advanced, sender payload and random sender reference
      * const uuidNamespace = '706ac453-2691-41af-9fde-ac5f787da1ec';
      * const paymentSubject = '...';
+     * const senderPayload = 'some sender payload';
+     * const payment = new nahmii.Payment(paymentAmount, senderWallet.address, recipientWallet.address, senderWallet, senderPayload);
+     *
+     * // Advanced, sender payload and semantic sender reference. See: https://github.com/uuidjs/uuid
+     * const uuidNamespace = '706ac453-2691-41af-9fde-ac5f787da1ec';
+     * const paymentSubject = '...';
+     * const senderPayload = 'some sender payload';
      * const senderRef = uuidv5(paymentSubject, uuidNamespace);
-     * const payment = new nahmii.Payment(paymentAmount, senderWallet.address, recipientWallet.address, senderWallet, senderRef);
+     * const payment = new nahmii.Payment(paymentAmount, senderWallet.address, recipientWallet.address, senderWallet, senderPayload, senderRef);
      */
-    constructor(amount, sender, recipient, walletOrProvider, senderRef) {
+    constructor(amount, sender, recipient, walletOrProvider, senderPayload, senderRef) {
         if (!(amount instanceof MonetaryAmount))
             throw new TypeError('amount is not an instance of MonetaryAmount');
 
+        if (undefined !== senderPayload && null !== senderPayload && 'string' !== typeof senderPayload)
+            throw new TypeError('senderPayload is not an instance of string');
+
         senderRef = senderRef || uuidv4();
 
         if (!/^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$/i.test(senderRef))
```

Comment:
Since this is an public API we should also add a size limit to senderPayload.
That will guard against storage abuse. Not sure what the right limit should be. 2000K?

Reply:
The size limit will be imposed in the trading API.